fitbenchmarking/parsing/cutest_parser.py

```diff
@@ -0,0 +1,240 @@
+        # In order to prevent the id being reused also store the x_data
+        self._cache = {id(fp.data_x): {'f': self._p.objcons, 'x': fp.data_x}}
+
+        return fp
+
+    def __del__(self):
+        self.mastsif_dir.cleanup()
+
+    def _import_problem(self, file_name):
+        """
+        Import the problem using cutest
+
+        :param file_name: The path to the file
+        :type file_name: str
+        :return: The parsed problem
+        :rtype: pycutest.CUTEstProblem
+        """
+
+        problem_ext = os.path.basename(file_name)
+        problem, _ = os.path.splitext(problem_ext)
+
+        return pycutest.import_problem(problem)
+
+    def _function(self, x, *params):
+        """
+        If these x values have been passed in before, then run the function
+        Otherwise, create a new problem file, parse and cache the function
+
+        :param x: The data to evaluate at
+        :type x: np.array
+        :return: The result of evaluating at the given x
+        :rtype: np.array
+        """
+        os.environ["MASTSIF"] = self.mastsif_dir.name
+
+        try:
+            f = self._cache[id(x)]['f']
+        except KeyError:
+            fname, _, _, _ = self._setup_data(x)
+            p = self._import_problem(fname)
+            f = p.objcons
+            self._cache[id(x)] = {'f': f, 'x': x}
+        _, fx = f(np.asarray(params))
+
+        return fx
+
+    def _get_starting_values(self):
+
+        starting_values = [
+            OrderedDict([
+                ('f{}'.format(i), self._p.x0[i])
+                for i in range(self._n)
+            ])
+        ]
+
+        return starting_values
+
+    def _setup_data(self, x=None):
+        """
+        Parses CUTEst SIF files to extract data points
+
+        :param x: X values to save (used for evaluating at new points)
+        :type x: np.array
+        :returns: data_x, data_y
+        :rtype: lists of floats
+        """
+
+        if self.file.closed:
+            with open(self._filename, 'r') as f:
+                lines = f.readlines()
+        else:
+            lines = self.file.readlines()
+
+        if x is not None and len(x.shape) == 0:
+            x = np.array([x])
+
+        x_idx, y_idx, e_idx = 0, 0, 0
+        data_x, data_y, data_e = None, None, None
+
+        # SIF requires columns of 25 chars
+        col_width = 25
+
+        written_x = False
+
+        to_write = []
+
+        for line in lines:
+
+            if "IE M " in line:
+                if x is None:
+                    self._m = int(line.split()[2])
+                    # this will always come before x/y data
+                    # so allocate space for these now
+                    data_x = np.zeros(self._m)
+                    data_y = np.zeros(self._m)
+                    data_e = np.zeros(self._m)
+                    # initialize index parameters for x and y
+                else:
+                    line = line[:col_width-1] + str(len(x))
```

Comment:
I could be wrong as I haven't paid especially close attention to the file format, however wouldn't you have to determine the amount of the line that you kept based on the length of the str you are adding to it e.g. `len(str(len(x)))`?

Reply:
As far as I'm aware it's tabulated columns that are left justified:
```
col1                     col2
a_value                  another_value
third_value              fourth
```
So it's fine to replace everything from that point onwards

Reply:
Ah my mistake, I mistook col_width for a line width

Reply:
There's a comment above where I define col_width that I can try to make clearer